Mathlib/ModelTheory/Substructures.lean

```diff
@@ -885,8 +887,295 @@ theorem range_subtype (S : L.Substructure M) : S.subtype.toHom.range = S := by
   rintro ⟨⟨y, hy⟩, rfl⟩
   exact hy
 
+variable (L) (M) (N)
+
+/-- An equivalence between substructures -/
+structure SubEquivalence  where
+  /--The substructure which is the domain of the equivalence.-/
+  sub_dom : L.Substructure M
+  /--The substructure which is the codomain of the equivalence.-/
+  sub_cod : L.Substructure N
+  /--The equivalence between the two subdomains.-/
+  equiv : sub_dom ≃[L] sub_cod
+
+@[inherit_doc]
+scoped[FirstOrder] notation:25 M " ≃ₚ[" L "] " N =>
+  FirstOrder.Language.Substructure.SubEquivalence L M N
```

Comment:
Apologies if this is obvious, but what is the `ₚ` in the notation standing for? "partial"?

Reply:
Yes - I'm thinking that these should be renamed `PartialEquiv`, and possibly other bits of nomenclature should be changed to match https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Equiv/PartialEquiv.html#PartialEquiv